date_range/wizard/date_range_generator.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import DAILY, MONTHLY, WEEKLY, YEARLY, rrule
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class DateRangeGenerator(models.TransientModel):
+    _name = 'date.range.generator'
+    _description = 'Date Range Generator'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get('date.range')
+
+    name_prefix = fields.Char('Range name prefix', required=True)
+    date_start = fields.Date(strint='Start date', required=True)
+    type_id = fields.Many2one(
+        comodel_name='date.range.type', string='Type', required=True,
+        domain="['|', ('company_id', '=', company_id), "
+               "('company_id', '=', False)]", ondelete='cascade')
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company',
+        default=_default_company)
+    unit_of_time = fields.Selection([
+        (str(YEARLY), 'years'),
+        (str(MONTHLY), 'months'),
+        (str(WEEKLY), 'weeks'),
+        (str(DAILY), 'days')], required=True)
```

Comment:
With 13.0 I think you must specify the default here:

```suggestion
        (str(DAILY), 'days')], required=True, default=str(YEARLY)))
```

Reply:
I think it's not needed.

Reply:
It's not needed for the module itself, but I got an error in the tests of `account_tax_balance` while migrating it (`account_tax_balance`  depends on this module). The error was not occurring in previous versions, so I guess they changed something in 13.0 that makes the definition of the default value necessary now.

Reply:
I think the problem is that for some reason you have records with this field empty, but the field is required. If not, I don't understand.